src-cpp/app/modules/shared/section_item.cpp

```diff
@@ -0,0 +1,156 @@
+#include <QDebug>
+
+#include "section_item.h"
+
+namespace Shared
+{
+namespace Models
+{
+SectionItem::SectionItem(QString id,
+                         SectionType sectionType,
+                         QString name,
+                         QString description,
+                         QString image,
+                         QString icon,
+                         QString color,
+                         bool active,
+                         bool enabled,
+                         bool amISectionAdmin,
+                         bool hasNotification,
+                         int notificationsCount,
+                         bool isMember,
+                         bool joined,
+                         bool canJoin,
+                         bool canManageUsers,
+                         bool canRequestAccess,
+                         int access,
+                         bool ensOnly):
+    m_id(id),
+    m_sectionType(sectionType) ,
+    m_name(name),
+    m_amISectionAdmin(amISectionAdmin),
+    m_description(description),
+    m_image(image),
+    m_icon(icon),
+    m_color(color),
+    m_hasNotification(hasNotification),
+    m_notificationsCount(notificationsCount),
+    m_active(active),
+    m_enabled(enabled),
+    m_isMember(isMember),
+    m_joined(joined),
+    m_canJoin(canJoin),
+    m_canManageUsers(canManageUsers),
+    m_canRequestAccess(canRequestAccess),
+    m_access(access),
+    m_ensOnly(ensOnly)
+{
+qDebug()<<" SECTION TYPE SET = "<< sectionType;
+}
+
+SectionType SectionItem::getSectionType()
+{
+    return m_sectionType;
+}
```

Comment:
can you explain what you mean a bit more?

Reply:
Sure, I am new to cpp so what I say might be stupid but I had the impression that we can do that in cpp:
```
class A
{
public:
   readonly int x;
}
```

So instead of declaring all the get accessor they are there for us thanks to the readonly
Makes sense?

Reply:
Hey no not at all, we dont have readonly in c++, we need to right getter and setters, so this is not possible.

Reply:
Alright then!

Reply:
that is not true :) first one:
```
class Test {
public:
    Test(int p): m_p(p) {}
private:
    const int m_p;
};
```

Reply:
and second:
```
SectionType SectionItem::getSectionType() const
{
    return m_sectionType;
}
```

Reply:
Hey what @alaibe mentioned was a readonly property without having the need to write getters, which means in the example you mentioned ->
```
class Test {
public:
    Test(int p): m_p(p) {}:
    const int m_p;
};
````
the m_p basically is public and not desirable and vulnerable ?